src/shared/actions/archives.js

```diff
@@ -18,25 +18,27 @@ export const removeArchive = createAction(ARCHIVES_REMOVE);
 export const clearArchives = createAction(ARCHIVES_CLEAR);
 
 export const loadArchive = payload => async (dispatch, getState) => {
-  const archive = await showPasswordDialog(
-    password => loadWorkspace(payload, password).catch(err => {
-      const unknownMessage = 'An unknown error has occurred';
-      return Promise.reject(
-        isError(err)
-          ? err.message || unknownMessage
-          : unknownMessage
-      );
-    })
-  );
+  try {
+    const archive = await showPasswordDialog(
+      password => loadWorkspace(payload, password).catch(err => {
+        const unknownMessage = 'An unknown error has occurred';
+        return Promise.reject(
```

Comment:
@perry-mitchell this one is for the password dialog component (swal) to handle and it wants a promise. the await thingie is for the result of that :D